launcher.py

```diff
@@ -222,58 +246,85 @@ def __init__(self):
 
         #Create checkboxes for each source option
         sources = self.modules.get("sources")
+        self.sources = sources
         up_source = [source.capitalize() for source in sources]
         self.src_dict = {source: i + 1 for i, source in enumerate(up_source)}
-        self.src_checkboxes = [QCheckBox(f"{src}") for src in up_source]
+        self.src_widgets = [(QCheckBox(f"{src}"), QLineEdit()) for src in up_source]
+        for checkbox, args in self.src_widgets:
+            args.setPlaceholderText("CLI args")
+        self.src_state = [[False, ""] for _ in sources]
+
         
         #Layout for source checkboxes
         self.src_layout = QGridLayout()
         row = 0
         col = 0
-        for checkbox in self.src_checkboxes:
+        for checkbox, args in self.src_widgets:
             self.src_layout.addWidget(checkbox, row, col)
+            self.src_layout.addWidget(args, row + 1, col)
             col += 1
             if col == 3:  # Three checkboxes per row
                 col = 0
-                row += 1
+                row += 2
 
         source_list = QWidget()
         source_list.setLayout(self.src_layout)
         source_list.setContentsMargins(0, 0, 0, 0)
 
+
         #Connect checkbox state to signals to detect which sources were selected 
-        for checkbox in self.src_checkboxes:
-            checkbox.stateChanged.connect(self.update_selected)
+        for i, (checkbox, args) in enumerate(self.src_widgets):
+            # https://docs.python.org/3/faq/programming.html#why-do-lambdas-defined-in-a-loop-with-different-values-all-return-the-same-result
+            def stateChanged(state, i=i):
```

Comment:
This is really small nitpick. Although the Python documentation officially suggests you can do this, I would be inclined to avoid this. The reason is because we don't know if `.connect` will always only give us one argument, and I assume that nobody really wants to take the time to check (and just looking at the [outdated] docs quickly is unclear https://doc.qt.io/qtforpython-5/PySide2/QtCore/Signal.html#PySide2.QtCore.Signal.connect). I think it's better to make sure it's clear that this is a function with arity 1 (i.e. takes exactly one argument).

Reply:
Hi :)) I didn't think about a good way to pass one argument because I have no idea how to edit state binding to a list. But I did use another solution with `partial` function to ensure it take two argument.